File: Source/StockObjects/Scripts/PathOperations.js
function PathOperations() {
}

PathOperations.prototype.GetContainerPath = function (path) {
    var pe = path.split("/");

    var result = "";
    for (var j = 0; j < pe.length - 1; ++j) {
        var e = pe[j];
        if (e.length > 0) result += "/" + e;
    }

    return result;
}

PathOperations.prototype.GetFileName = function (path) {
    if (!path) return "";
    var pe = path.split("/");

    if (pe.length > 1)
    {
        return pe[pe.length - 1];
    }
    else {
        return "";
    }
}

PathOperations.prototype.IsPrefixOf = function (tobetested, ofwhat) {
    var pet = tobetested ? tobetested.split("/") : [];
    var pew = ofwhat && ofwhat != "/" ? ofwhat.split("/") : [];

    pet = pet.filter(function (e) { return e != ""; });
    pew = pew.filter(function (e) { return e != ""; });

    var n = pew.length;
    if (pet.length < n) return false;

    for (var j = 0; j < n; ++j)
    {
        if (pew[j] != pet[j]) return false;
    }

    return true;
}

PathOperations.prototype.ComposeURL = function (url, subpath) {
    var k = url.length;

    // make sure url is terminated with a slash
    if (k > 0) {
        if (url[k - 1] != "/") {
            url = url + "/";
        }
    }

    // append unprefix subpath
    var pe = subpath.split("/");
    return url + pe.join("/");
}

File: Source/StockObjects/PathOperations.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Wpf2Html5.StockObjects
{
    [GeneratorIgnore]
    public class PathOperations
    {
        public string GetContainerPath(string path)
        {
            throw new NotImplementedException();
        }

        public string CombinePath(string container, string relativepath)
        {
            throw new NotImplementedException();
        }

        public string GetFileName(string CurrentPath)
        {
            throw new NotImplementedException();
        }

        public bool IsPrefixOf(string tobetested, string ofwhat)
        {
            throw new NotImplementedException();
        }

        public string ComposeURL(string url, string path)
        {
            if(!url.EndsWith("/") && !path.StartsWith("/"))
            {
                return url + "/" + path;
            }
            else
            {
                return url + path;
            }
        }
    }
}
